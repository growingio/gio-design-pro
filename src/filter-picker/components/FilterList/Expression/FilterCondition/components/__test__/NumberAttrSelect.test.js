import React from 'react';
import { shallow, mount } from 'enzyme';
import NumberAttrSelect from '../NumberAttrSelect.tsx';

describe('<NumberAttrSelect />', () => {
  it('should throw error', () => {
    expect(() => shallow(<NumberAttrSelect />)).toThrowError();
  });

  it('attrSelect = "="', () => {
    let wrapper = shallow(<NumberAttrSelect values={[]} attrSelect="=" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
    wrapper = shallow(<NumberAttrSelect values={['10']} attrSelect="=" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('attrSelect = "between"', () => {
    let wrapper = shallow(<NumberAttrSelect values={[]} attrSelect="between" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
    wrapper = shallow(<NumberAttrSelect values={['30']} attrSelect="between" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
    wrapper = shallow(<NumberAttrSelect values={['-30', '40']} attrSelect="between" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('attrSelect = "hasValue"', () => {
    const wrapper = shallow(<NumberAttrSelect values={[]} attrSelect="hasValue" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('attrSelect = "noValue"', () => {
    const wrapper = shallow(<NumberAttrSelect values={[]} attrSelect="noValue" attrChange={() => {}} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('default change', async () => {
    let value = '';
    const wrapper = mount(
      <NumberAttrSelect
        values={[10]}
        attrSelect="="
        attrChange={(val) => {
          value = val;
        }}
      />
    );
    wrapper.find('input').simulate('focus');
    wrapper.find('input').simulate('change', { target: { value: '2' } });
    expect(value[0]).toBe(2);
    wrapper.find('input').simulate('change', { target: { value: '' } });
    expect(value[0]).toBe('0');
  });

  it('between change', () => {
    let value = [];
    const wrapper = mount(
      <NumberAttrSelect
        values={[]}
        attrSelect="between"
        attrChange={(val) => {
          value = val;
        }}
      />
    );
    wrapper.find('input').at(0).simulate('focus');
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: '-2' } });
    expect(value[0]).toBe(-2);

    wrapper
      .find('input')
      .at(0)
      .simulate('change', { target: { value: '' } });
    expect(value[0]).toBe('0');

    wrapper.find('input').at(1).simulate('focus');
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { target: { value: '20' } });
    expect(value[1]).toBe(20);
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { target: { value: '' } });
    expect(value[1]).toBe('0');
  });
});
