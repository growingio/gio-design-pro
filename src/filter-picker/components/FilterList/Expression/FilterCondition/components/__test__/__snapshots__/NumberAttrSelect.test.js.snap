// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberAttrSelect /> attrSelect = "=" 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "gio-input gio-input--middle",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "gio-input__content",
          "placeholder": "",
          "type": "text",
          "value": "0",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<NumberAttrSelect /> attrSelect = "=" 2`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "gio-input gio-input--middle",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "gio-input__content",
          "placeholder": "",
          "type": "text",
          "value": "10",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<NumberAttrSelect /> attrSelect = "between" 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "style": "display:flex;align-items:center",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
          "style": "width:120px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "",
              "type": "text",
              "value": "0",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "style": "margin:0 16px",
          },
          "children": Array [
            Node {
              "data": "与",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "gio-input gio-input--middle",
              "style": "width:120px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "gio-input__content",
                  "placeholder": "",
                  "type": "text",
                  "value": "0",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "style": "margin:0 16px",
        },
        "children": Array [
          Node {
            "data": "与",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
            "style": "width:120px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "",
                "type": "text",
                "value": "0",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
            "style": "width:120px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "",
                "type": "text",
                "value": "0",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
          "style": "width:120px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "",
              "type": "text",
              "value": "0",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "style": "margin:0 16px",
          },
          "children": Array [
            Node {
              "data": "与",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "gio-input gio-input--middle",
              "style": "width:120px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "gio-input__content",
                  "placeholder": "",
                  "type": "text",
                  "value": "0",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<NumberAttrSelect /> attrSelect = "between" 2`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "style": "display:flex;align-items:center",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
          "style": "width:120px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "",
              "type": "text",
              "value": "30",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "style": "margin:0 16px",
          },
          "children": Array [
            Node {
              "data": "与",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "gio-input gio-input--middle",
              "style": "width:120px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "gio-input__content",
                  "placeholder": "",
                  "type": "text",
                  "value": "0",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "style": "margin:0 16px",
        },
        "children": Array [
          Node {
            "data": "与",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
            "style": "width:120px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "",
                "type": "text",
                "value": "0",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
            "style": "width:120px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "",
                "type": "text",
                "value": "30",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
          "style": "width:120px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "",
              "type": "text",
              "value": "0",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "style": "margin:0 16px",
          },
          "children": Array [
            Node {
              "data": "与",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "gio-input gio-input--middle",
              "style": "width:120px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "gio-input__content",
                  "placeholder": "",
                  "type": "text",
                  "value": "30",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<NumberAttrSelect /> attrSelect = "between" 3`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "style": "display:flex;align-items:center",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
          "style": "width:120px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "",
              "type": "text",
              "value": "-30",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "style": "margin:0 16px",
          },
          "children": Array [
            Node {
              "data": "与",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Node {
            "attribs": Object {
              "class": "gio-input gio-input--middle",
              "style": "width:120px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "gio-input__content",
                  "placeholder": "",
                  "type": "text",
                  "value": "40",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "style": "margin:0 16px",
        },
        "children": Array [
          Node {
            "data": "与",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
            "style": "width:120px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "",
                "type": "text",
                "value": "40",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
            "style": "width:120px",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "",
                "type": "text",
                "value": "-30",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
          "style": "width:120px",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "",
              "type": "text",
              "value": "40",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "style": "margin:0 16px",
          },
          "children": Array [
            Node {
              "data": "与",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Node {
            "attribs": Object {
              "class": "gio-input gio-input--middle",
              "style": "width:120px",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "gio-input__content",
                  "placeholder": "",
                  "type": "text",
                  "value": "-30",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "placeholder": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<NumberAttrSelect /> attrSelect = "hasValue" 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`<NumberAttrSelect /> attrSelect = "noValue" 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
